### DISCUSSION TITLE
Field selection dependency

### DISCUSSION THREAD

**Discussion by JakeSmokie** [ChilliCream/hotchocolate#4850] [Feature Request]
Is there a way to add a dependency for some field?
This example returns `null` if `descriptionLocaleId` is not selected in gql query.

```cs
descriptor.Field("description").Resolver(ctx => ctx.Parent<Item>().DescriptionLocaleId);
```

I am using EF Core 5, Hot Chocolate v11 with `UseProjectionAttribute`

Desired functionality:
```cs
descriptor.Field("description").DependsOn(x => x.DescriptionLocaleId).Resolver(ctx => ctx.Parent<Item>().DescriptionLocaleId);
```

**Comment by PascalSenn** [comment]
@JakeSmokie 
We plan to add resolver metadata to do exactly that

For now you can use the IsProjectedAttribute 
```csharp
[IsProjected(true)]
public string DescriptionLocaleId {get;set;}
```
or set it via
```csharp 
descriptor.Field(x => x.DescriptionLocaleId).IsProjected(true);
```

This way this field is always projected and your resolver should work. 
Direct resolver dependencies will come in a later release

**Comment by Sonic198** [comment]
Any plans when that feature could be added?

**Comment by nikolai-mb** [comment]
Also waiting for this feature, would be a great addition.

**Comment by evTop** [comment]
Is there any further progress on this feature?

**Comment by celadrian6** [comment]
I also am looking for this feature :)
